### PR TITLE
feat: send jobs per e-mail

### LINKED ISSUE #29: Email results of the job search
## Description of feature 
Email contents of the job listings file `job_search.txt` to a specified email address
Sender and receiver email address should be configurable in a config file

### PR DESCRIPTION
**Pull request by AzizCode92**
### 2. Description
Add configuration file.
Send jobs description via gmail.

### 3. Fill in checklist by marking [x]

- [x] I've read the CONTRIBUTING.md
- [ ] I was assigned to this issue
- [x] My code is formatted using required linting
- [x] I've run my code locally and checked it works

### CODE REVIEW

**Comment by funbeedev** [comment]
Thanks for your PR. We'll review this soon!

**Comment by funbeedev** [comment]
@AzizCode92 I've checked your code locally but it doesn't appear to be working for me. 
When I configure my gmail details and run the script I get the following error:
```Traceback (most recent call last):
  File "job-search-web-scraping.py", line 64, in <module>
    indeed_job_search(*sys.argv)
  File "job-search-web-scraping.py", line 53, in indeed_job_search
    gmailer.send(sender=cfg["sender_email"],
  File "/home/JobSearchWebScraping/utils/mail.py", line 44, in send
    msg = MIMEMultipart.MIMEMultipart()
AttributeError: type object 'MIMEMultipart' has no attribute 'MIMEMultipart'
```
Also, could you explain why the `utils/` folder is in the directory of the script? Ideally we want a way of sending emails without having to include source code of the module.
There are built in python modules called  `smtplib` and `email.message` which you can use to send emails from any mail client and not just Gmail.

You can import them using:
```
import smtplib
from email.message import EmailMessage
```
Can you modify your PR to use these modules instead? It will allow us to send emails from any mail client.
I also have a post on my blog explaining how to send emails using these modules if you want to use that as a guideline:
https://inspirezone.tech/sending-emails-with-python/

If you have any questions let me know!

**Comment by AzizCode92** [comment]
Hi @funbeedev, I will look into your response asap. 
thanks.

**Comment by AzizCode92** [comment]
I got an error when trying to execute the code locally, 
`selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"[name="q"]"}
`
Unfortunately, I wasn't able to verify wether the code works locally or not, if it works fine for you let me know.

thanks.

**Comment by funbeedev** [comment]
@AzizCode92 Indeed.com changed the layout of their site a bit which is what is causing the error you see when you run the script locally. 
The problem is the URL `https://www.indeed.com/worldwide` no longer has a search field. Change this to `https://www.indeed.com` and the enter a city in your country for the search to work. 

After I made these changes the job search will work but the email functionality is not working. Could you change the URL as mentioned above and add a relevant city so you can test your changes locally and make sure it works. Thanks. 
This is the error I'm getting now:
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/email/message.py", line 208, in attach
    self._payload.append(payload)
AttributeError: 'str' object has no attribute 'append'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "job-search-web-scraping.py", line 106, in <module>
    indeed_job_search(*sys.argv)
  File "job-search-web-scraping.py", line 97, in indeed_job_search
    send_email(cfg["sender_email"], cfg["sender_password"],
  File "job-search-web-scraping.py", line 38, in send_email
    message.attach(part)
  File "/usr/lib/python3.8/email/message.py", line 210, in attach
    raise TypeError("Attach is not valid on a message with a"
TypeError: Attach is not valid on a message with a non-multipart payload

```

**Comment by AzizCode92** [comment]
Hi @funbeedev, I tested the code and it worked for me locally.
try it and let me know if everything is ok for you as well.
Bests,
Aziz

File: job-search-web-scraping.py [hunk 1]
```diff
@@ -1,39 +1,88 @@
...
+    server.quit()
 
 
 def indeed_job_search(*args):
-
     browser = None
 
     PATH_TO_GECKO_DRIVER = './geckodriver'
-    PATH_TO_CHROME_DRIVER = './chromedriver'
+    PATH_TO_CHROME_DRIVER = '/usr/local/bin/chromedriver'
```
Comment (funbeedev) [inline job-search-web-scraping.py:60]: Can you change this to the original line 
`PATH_TO_CHROME_DRIVER = './chromedriver'`

File: job-search-web-scraping.py [hunk 2]
```diff
@@ -1,39 +1,88 @@
...
 
     if Path(PATH_TO_GECKO_DRIVER).is_file():
         options = webdriver.FirefoxOptions()
         if 'headless' in args:
             options.headless = True
-        browser = webdriver.Firefox(executable_path=PATH_TO_GECKO_DRIVER, options=options)
+        browser = webdriver.Firefox(executable_path=PATH_TO_GECKO_DRIVER,
+                                    options=options)
```
Comment (funbeedev) [inline job-search-web-scraping.py:67]: Can you reset the formatting of lines that were made shorter from the original source? There's no need to worry about the flake8 error ` E501 line too long`. 
Change this line and all other lines that were made shorter. Thanks.

File: job-search-web-scraping.py [hunk 3]
```diff
@@ -1,39 +1,88 @@
...
+               email_body, file_path):
+    """
+    sender_email_address: sender email address
+    email_password: sender email password
+    receiver_email_address: receiver email password
+    email_subject: email subject
+    email_body: email body
+    file_path: file to attach in the email
+    """
+    email_smtp = "smtp.gmail.com"
```
Comment (funbeedev) [inline job-search-web-scraping.py:26]: The email_smtp can be made configurable in the config file. This will allow other mail servers to be used.

**Review by funbeedev — COMMENTED** [review]
@AzizCode92 
The code review comments described changes to be made before merging. Thanks.

**Comment by funbeedev** [comment]
@AzizCode92 I checked the script and it works now! Just a few changes to be made as described in my code review comments. Thanks.

**Comment by funbeedev** [comment]
@AzizCode92  Thanks for making the changes. I made a slight edit to the config file to make all the options blank. All looks good now so I will merge. Thanks for your contribution!

**Pull request merged**